src/msg/simple/Pipe.cc

```diff
@@ -2244,7 +2257,7 @@ int Pipe::do_sendmsg(struct msghdr *msg, unsigned len, bool more)
     }
   }
   restore_sigpipe();
-  return 0;
+  return rc;
```

Comment:
why you are changing this line? imo, it would be easier for the reader if we are sure that this branch always return `0`?